File: java/src/main/java/ps_traning/programmers/level_2/카펫.java
package ps_traning.programmers.level_2;

public class 카펫 {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        for (int c = 1; c <= (int) Math.sqrt(yellow); c++) {
            if (yellow % c != 0) {
                continue;
            }
            int r = yellow / c;
            if (r * 2 + c * 2 + 4 == brown) {
                answer = new int[]{r + 2, c + 2};
                break;
            }
        }
        return answer;
    }
}

File: python/with_java/programmers/level 2/카펫.py
def solution(brown, yellow):
    answer = []
    for c in range(1, int(yellow ** 0.5)+1):
        if yellow % c != 0:
            continue
        r = yellow // c
        if r*2 + c*2 + 4 == brown:
            answer = [r+2, c+2]
            break
    return answer